### PR TITLE
Add the functionality to use Bing image of the day in the Wallpaper section 

### PR DESCRIPTION
**Pull request by YoussefLasheen**
I modified the the wallpaper_model file to allow for adding 3 options for wallpaper mode. 
Then Added the refreshBingWallpaper function to get today's image url then download it then save it to the documents folder.
I also added a refresh button if the image image download wasn't triggered automatically.

I added the http package to download the image.

Ref #189

### CODE REVIEW

**Comment by Feichtmeier** [comment]
Amazing! Works very well

Bildschirmfoto von 2021-12-26 11-29-38

Adding some inline nit-picks for the review

File: lib/view/pages/wallpaper/wallpaper_model.dart [hunk 1]
```diff
@@ -119,7 +131,32 @@ class WallpaperModel extends SafeChangeNotifier {
     pictureUri = list.first;
   }
 
-  bool get isColorBackground => pictureUri.isEmpty ? true : false;
+  Future<void> refreshBingWallpaper() async {
+    Future<String> getBingImageUrl() async {
+      const address =
+          'https://www.bing.com/HPImageArchive.aspx?format=js&idx=0&n=1&mkt=en-US';
```
Comment (Feichtmeier) [inline lib/view/pages/wallpaper/wallpaper_model.dart:137]: I think this URL should be stored as a constant somewhere. Maybe just in the model?

File: lib/view/pages/wallpaper/wallpaper_model.dart [hunk 2]
```diff
@@ -119,7 +131,32 @@ class WallpaperModel extends SafeChangeNotifier {
...
+      http.Response imageMetadataResponse = await http.get(Uri.parse(address));
+      return 'http://www.bing.com' +
```
Comment (Feichtmeier) [inline lib/view/pages/wallpaper/wallpaper_model.dart:139]: You could move this bing "präfix" to some constant, too

**Review by Feichtmeier — CHANGES_REQUESTED** [review]

File: lib/view/pages/wallpaper/wallpaper_model.dart (hunk 1 above)
Comment (YoussefLasheen) [reply lib/view/pages/wallpaper/wallpaper_model.dart:137]: If I understand you correctly. The url can't be const as it changes everyday. 
I can make the whole getBingImageUrl a static function if that helps.

**Comment by YoussefLasheen** [comment]
I have made another commit. Did it solve the problem?

**Review by Feichtmeier — APPROVED** [review]

**Comment by Feichtmeier** [comment]
tyvm!

**Pull request merged**